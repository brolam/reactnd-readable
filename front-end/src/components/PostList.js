import React from 'react'
import PropTypes from 'prop-types'
import OrderOptions from '../components/OrderOptions'
import PostItem from '../components/PostItem'

function PostList({ 
  posts, 
  onSelectedPost = (posts) => { },
  onVoteScorePost = (postId, option) => { } }) {
  return (
    <div className="posts">
      <div className="posts-title">
        <span>Posts</span>
        <OrderOptions />
      </div>
      <div className="posts-list">
        {posts.map(post => {
          return (
            <PostItem
              key={post.id}
              post={post}
              onSelected={ (e) => onSelectedPost(post)}
              onVoteScorePost={onVoteScorePost}
            />
          )
        })}
      </div>
    </div>
  )
}

PostList.propTypes = {
  posts: PropTypes.array.isRequired,
}

export default PostList
