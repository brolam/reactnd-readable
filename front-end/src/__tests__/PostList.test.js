import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import PostList from '../components/PostList'


test('render without error', () => {
  ReactDOM.render(
    <PostList posts={posts} />,
    document.createElement('div'));
})

test('last Snapshot', () => {
  const postList = renderer.create(<PostList posts={posts} />);
  let tree = postList.toJSON();
  expect(tree).toMatchSnapshot();
})

const posts = [{
  category: "udacity",
  id: "7ni6ok3ym7mf1p33lnez",
  title: "Udacity is the best place to learn technology.",
  timestamp: 1467166872634,
  author: "thingtwo",
  body: "Everyone says so after all.",
  deleted: false,
  voteScore: 5
},
{
  category: "react",
  id: "8xf0y6ziyjabvozdd253nd",
  title: "Udacity is the best place to learn React",
  timestamp: 1467166872634,
  author: "thingtwo",
  body: "Everyone says so after all.",
  deleted: false,
  voteScore: 5
}]
