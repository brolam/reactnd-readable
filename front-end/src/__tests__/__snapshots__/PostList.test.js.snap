// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`last Snapshot 1`] = `
<div
  className="posts"
>
  <div
    className="posts-title"
  >
    <span>
      Posts
    </span>
    <div
      className="order-options"
    >
      <select
        defaultValue="voteScore"
        onChange={[Function]}
      >
        <option
          value="voteScore"
        >
          Vote Score
        </option>
        <option
          value="publishedDate"
        >
          Published date
        </option>
      </select>
    </div>
  </div>
  <div
    className="posts-list"
  >
    <div
      className="post"
      onClick={[Function]}
    >
      <div
        className="post-header"
      >
        <div
          className="post-header-title"
        >
          <span
            className="category udacity"
          >
            Udacity
          </span>
           
          Udacity is the best place to learn technology.
        </div>
        <div
          className="post-header-published"
        >
          <span
            className="published"
          >
            a year ago
          </span>
        </div>
      </div>
      <div
        className="post-header-author"
      >
        by 
        thingtwo
      </div>
      <div
        className="post-body"
      >
        Everyone says so after all.
      </div>
      <div
        className="post-footer"
      >
        <button
          className="edit-button"
          href="/"
          onClick={[Function]}
        >
          Edit
        </button>
        <button
          className="delete-button"
          onClick={[Function]}
        >
          Delete
        </button>
        <button
          className="liked"
          onClick={[Function]}
        >
          Liked 
          <span>
            5
          </span>
        </button>
        <button
          className="not-liked"
          onClick={[Function]}
        >
          Not like
        </button>
      </div>
    </div>
    <div
      className="post"
      onClick={[Function]}
    >
      <div
        className="post-header"
      >
        <div
          className="post-header-title"
        >
          <span
            className="category react"
          >
            React
          </span>
           
          Udacity is the best place to learn React
        </div>
        <div
          className="post-header-published"
        >
          <span
            className="published"
          >
            a year ago
          </span>
        </div>
      </div>
      <div
        className="post-header-author"
      >
        by 
        thingtwo
      </div>
      <div
        className="post-body"
      >
        Everyone says so after all.
      </div>
      <div
        className="post-footer"
      >
        <button
          className="edit-button"
          href="/"
          onClick={[Function]}
        >
          Edit
        </button>
        <button
          className="delete-button"
          onClick={[Function]}
        >
          Delete
        </button>
        <button
          className="liked"
          onClick={[Function]}
        >
          Liked 
          <span>
            5
          </span>
        </button>
        <button
          className="not-liked"
          onClick={[Function]}
        >
          Not like
        </button>
      </div>
    </div>
  </div>
</div>
`;
